cpp/CMakeLists.txt

```diff
@@ -100,15 +102,28 @@ endif((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND(CMAKE_CXX_COMPILER_VERSION VERS
 
 # define flags to enable most warnings and treat them as errors for different compilers
 # add flags to each target separately instead of globally so users have the choice to use their own flags
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-    set(MEMILIO_CXX_FLAGS_ENABLE_WARNING_ERRORS
-        "-Wno-unknown-warning;-Wno-pragmas;-Wall;-Wextra;-Werror;-Wshadow;--pedantic-errors;-Wno-deprecated-copy;-Wno-expansion-to-defined")
-elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    set(MEMILIO_CXX_FLAGS_ENABLE_WARNING_ERRORS
-        "-Wno-unknown-warning-option;-Wall;-Wextra;-Werror;-Wshadow;--pedantic-errors;-Wno-deprecated;-Wno-gnu-zero-variadic-macro-arguments")
-elseif(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-    set(MEMILIO_CXX_FLAGS_ENABLE_WARNING_ERRORS
-        "/W4;/WX")
+if(MEMILIO_ENABLE_WARNINGS_AS_ERRORS)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+        set(MEMILIO_CXX_FLAGS_ENABLE_WARNING_ERRORS
+            "-Wno-unknown-warning;-Wno-pragmas;-Wall;-Wextra;-Werror;-Wshadow;--pedantic-errors;-Wno-deprecated-copy;-Wno-expansion-to-defined")
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        set(MEMILIO_CXX_FLAGS_ENABLE_WARNING_ERRORS
+            "-Wno-unknown-warning-option;-Wall;-Wextra;-Werror;-Wshadow;--pedantic-errors;-Wno-deprecated;-Wno-gnu-zero-variadic-macro-arguments")
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+        set(MEMILIO_CXX_FLAGS_ENABLE_WARNING_ERRORS
+            "/W4;/WX")
+    endif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")

```

Comment:
old cmake required `else` and `endif` to repeat the condition. Luckily, this isn't needed anymore. I think it is still in our code in a few places, but it's ugly and unnecessary, so please just use `else()` and `endif()`